qcodes/instrument_drivers/AlazarTech/ATS.py

```diff
@@ -864,11 +857,17 @@ def get_sample_rate(self):
         Returns:
             the number of samples (per channel) per second
         """
-        if self.sample_rate.get() == 'EXTERNAL_CLOCK':
+        if (self.clock_source.get() == 'EXTERNAL_CLOCK_10MHz_REF'
```

Comment:
where did the decimation part of this function go?

Reply:
furthermore, is th elogic here correct? when there is an external clock, then the sample rate is still determined by the internal clock setting I think

Reply:
The decimation is handled below like its always been.

If clock_source == 'EXTERNAL_CLOCK_10MHz_REF' and there is a external_sample_rate parameter it will use that. Alternatively it will fall back to the internal sample rate if that parameter is not defined. That is to accommodate for the 2 different Alazar Cards. One has a freely settable sample rate with a 10 MHZ external clock and the other one only has one possible rate in that case.

If there is a pure external_clock there is still no automatic way to infer the sample rate.